homeassistant/components/dyson/fan.py

```diff
@@ -239,6 +324,256 @@ def supported_features(self) -> int:
+                     FanSpeed.FAN_SPEED_2.value: SPEED_LOW,
+                     FanSpeed.FAN_SPEED_3.value: SPEED_LOW,
+                     FanSpeed.FAN_SPEED_4.value: SPEED_LOW,
+                     FanSpeed.FAN_SPEED_AUTO.value: SPEED_MEDIUM,
+                     FanSpeed.FAN_SPEED_5.value: SPEED_MEDIUM,
+                     FanSpeed.FAN_SPEED_6.value: SPEED_MEDIUM,
+                     FanSpeed.FAN_SPEED_7.value: SPEED_MEDIUM,
+                     FanSpeed.FAN_SPEED_8.value: SPEED_HIGH,
+                     FanSpeed.FAN_SPEED_9.value: SPEED_HIGH}
+
+        return speed_map[self._device.state.speed]
+
+    @property
+    def dyson_speed(self):
+        """Return the current speed."""
+        from libpurecool.const import FanSpeed
+
+        if self._device.state:
+            if self._device.state.speed == FanSpeed.FAN_SPEED_AUTO.value:
+                return self._device.state.speed
+            return int(self._device.state.speed)
+
+    @property
+    def night_mode(self):
+        """Return Night mode."""
+        return self._device.state.night_mode == "ON"
+
+    @property
+    def auto_mode(self):
+        """Return Auto mode."""
+        return self._device.state.auto_mode == "ON"
+
+    @property
+    def angle_low(self):
+        """Return angle high."""
+        return int(self._device.state.oscillation_angle_low)
+
+    @property
+    def angle_high(self):
+        """Return angle low."""
+        return int(self._device.state.oscillation_angle_high)
+
+    @property
+    def flow_direction_front(self):
+        """Return frontal flow direction."""
+        return self._device.state.front_direction == 'ON'
+
+    @property
+    def timer(self):
+        """Return timer."""
+        return self._device.state.sleep_timer
+
+    @property
+    def hepa_filter(self):
+        """Return the HEPA filter state."""
+        return int(self._device.state.hepa_filter_state)
+
+    @property
+    def carbon_filter(self):
+        """Return the carbon filter state."""
+        return int(self._device.state.carbon_filter_state)
+
+    @property
+    def speed_list(self) -> list:
+        """Get the list of available speeds."""
+        return [SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
+
+    @property
+    def dyson_speed_list(self) -> list:
+        """Get the list of available dyson speeds."""
+        from libpurecool.const import FanSpeed
+        return [
+            int(FanSpeed.FAN_SPEED_1.value),
+            int(FanSpeed.FAN_SPEED_2.value),
+            int(FanSpeed.FAN_SPEED_3.value),
+            int(FanSpeed.FAN_SPEED_4.value),
+            int(FanSpeed.FAN_SPEED_5.value),
+            int(FanSpeed.FAN_SPEED_6.value),
+            int(FanSpeed.FAN_SPEED_7.value),
+            int(FanSpeed.FAN_SPEED_8.value),
+            int(FanSpeed.FAN_SPEED_9.value),
+            int(FanSpeed.FAN_SPEED_10.value),
+        ]
+
+    @property
+    def device_serial(self):
+        """return fan's serial number."""
```

Comment:
Capitalize the first word in the docstring.

Reply:
Done.